src/compiler/checker.ts

```diff
@@ -13520,6 +13521,34 @@ namespace ts {
             return instantiation;
         }
 
+        function getLocalTypeAliasInstantiation(symbol: Symbol, typeArguments: readonly Type[] | undefined, outerTypeParameterMapper: TypeMapper | undefined, aliasSymbol?: Symbol, aliasTypeArguments?: readonly Type[]): Type {
+            const type = getDeclaredTypeOfSymbol(symbol);
+            if (type === intrinsicMarkerType && intrinsicTypeKinds.has(symbol.escapedName as string) && typeArguments && typeArguments.length === 1) {
+                return getStringMappingType(symbol, typeArguments[0]);
+            }
+            const links = getSymbolLinks(symbol);
+            const outerTypeParameters = getOuterTypeParametersOfAlias(symbol);
+            const mappedOuterParams = outerTypeParameterMapper && outerTypeParameters && instantiateTypes(outerTypeParameters, outerTypeParameterMapper);
+            const typeParameters = links.typeParameters!;
+            const id = getTypeListId(concatenate(mappedOuterParams, typeArguments)) + getAliasId(aliasSymbol, aliasTypeArguments);
+            let instantiation = links.instantiations!.get(id);
+            if (!instantiation) {
+                const outerTypeArgumentMapper = outerTypeParameters && mappedOuterParams && createTypeMapper(outerTypeParameters, mappedOuterParams);

```

Comment:
do you need to check `outerTypeParameters` again here? It's already checked as part of `mappedOuterParams`

Reply:
For control flow to be happy, yes. XD